File: src/main/java/com/bot/commands/voice/ListTracksCommand.java
package com.bot.commands.voice;

import com.bot.commands.VoiceCommand;
import com.bot.voice.QueuedAudioTrack;
import com.bot.voice.VoiceSendHandler;
import com.jagrosh.jdautilities.command.CommandEvent;
import datadog.trace.api.Trace;

import java.util.ArrayList;
import java.util.List;

public class ListTracksCommand extends VoiceCommand {

	public ListTracksCommand() {
		this.name = "list";
		this.arguments = "";
		this.help = "Lists the tracks currently in the queue";
		this.aliases = new String[]{"playlist", "lists", "queue"} ;
	}

	@Override
	//@trace(operationName = "executeCommand", resourceName = "ListTracks")
	protected void executeCommand(CommandEvent commandEvent) {
		VoiceSendHandler handler = (VoiceSendHandler) commandEvent.getGuild().getAudioManager().getSendingHandler();

		if (handler == null) {
			commandEvent.reply(commandEvent.getClient().getWarning() + " I am not currently playing audio.");
			return;
		}

		if (!handler.isPlaying()) {
			commandEvent.reply(commandEvent.getClient().getWarning() + " I am not currently playing audio.");
			return;
		}

		QueuedAudioTrack nowPlaying = handler.getNowPlaying();
		List<QueuedAudioTrack> tracks = new ArrayList<>(handler.getTracks());
		StringBuilder sb = new StringBuilder();
		sb.append("```\nNow Playing: " + nowPlaying.getTrack().getInfo().title + "\n");

		for (int i = 0; i < tracks.size(); i++) {
			if (i == 0)
				sb.append("Next: " + tracks.get(i).getTrack().getInfo().title + "\n");
			else
				sb.append((i + 1) + ": " + tracks.get(i).getTrack().getInfo().title + "\n");
		}
		sb.append("```");
		commandEvent.reply(sb.toString());
	}
}

File: src/main/java/com/bot/commands/voice/ListTracksCommand.kt
package com.bot.commands.voice

import com.bot.commands.VoiceCommand
import com.bot.voice.QueuedAudioTrack
import com.bot.voice.VoiceSendHandler
import com.jagrosh.jdautilities.command.CommandEvent
import com.jagrosh.jdautilities.commons.waiter.EventWaiter
import com.jagrosh.jdautilities.menu.Paginator
import java.util.*
import java.util.concurrent.TimeUnit

class ListTracksCommand(waiter: EventWaiter) : VoiceCommand() {
    private val builder: Paginator.Builder

    //@trace(operationName = "executeCommand", resourceName = "ListTracks")
    override fun executeCommand(commandEvent: CommandEvent) {
        val handler = commandEvent.guild.audioManager.sendingHandler as VoiceSendHandler?
        if (handler == null) {
            commandEvent.reply(commandEvent.client.warning + " I am not currently playing audio.")
            return
        }
        if (!handler.isPlaying) {
            commandEvent.reply(commandEvent.client.warning + " I am not currently playing audio.")
            return
        }
        val nowPlaying = handler.nowPlaying
        val tracks: List<QueuedAudioTrack> = ArrayList(handler.tracks)
        val trackNames = ArrayList<String>()

        trackNames.add("Now Playing: ${nowPlaying.track.info.title}")

        tracks.forEach {
            trackNames.add("${trackNames.size}: ${it.track.info.title}")
        }
        builder.setText("Tracks in the queue")
                .setColor(commandEvent.selfMember.color)
        builder.setItems(*trackNames.toTypedArray())
        builder.build().paginate(commandEvent.channel, 1)
    }

    init {
        name = "list"
        aliases = arrayOf("playlist", "lists", "queue")
        builder = Paginator.Builder()
                .setColumns(1)
                .setItemsPerPage(15)
                .useNumberedItems(false)
                .showPageNumbers(true)
                .setEventWaiter(waiter)
                .setTimeout(30, TimeUnit.SECONDS)
                .waitOnSinglePage(false)
                .setFinalAction { message -> message.clearReactions().queue() }
    }
}